### PR TITLE
Avoid guzzle for discovering built-in CODE server

### PR DESCRIPTION
**Pull request by mrkara**
Signed-off-by: Muhammet Kara <muhammet.kara@collabora.com>

### CODE REVIEW

File: lib/Controller/DocumentController.php [hunk 1]
```diff
@@ -157,8 +157,21 @@ private function getDiscovery() {
...
+				// If we are sending query to built-in CODE server, we avoid using IClint::get() method
+				// because of an encoding issue in guzzle: https://github.com/guzzle/guzzle/issues/1758
+				if (strpos($wopiDiscovery, 'proxy.php') === false) {
+					$wopiClient = \OC::$server->getHTTPClientService()->newClient();
+					$discovery = $wopiClient->get($wopiDiscovery)->getBody();
+				}
+				else {
+					$arrContextOptions=array(
+						"ssl"=>array(
+							"verify_peer"=>false,
```
Comment (DeepDiver1975) [inline lib/Controller/DocumentController.php:169]: This should not be used in a live deployment - this is a potential sec risk

Comment (mrkara) [reply lib/Controller/DocumentController.php]: > This should not be used in a live deployment - this is a potential sec risk

Thank you for the review! :)

I have adjusted the patch to enable verify_peer, and set allow_self_signed to true. That is common for test/local/small installations.

File: lib/Controller/DocumentController.php [hunk 2]
```diff
@@ -157,8 +157,21 @@ private function getDiscovery() {
...
+				// because of an encoding issue in guzzle: https://github.com/guzzle/guzzle/issues/1758
+				if (strpos($wopiDiscovery, 'proxy.php') === false) {
+					$wopiClient = \OC::$server->getHTTPClientService()->newClient();
+					$discovery = $wopiClient->get($wopiDiscovery)->getBody();
+				}
+				else {
+					$arrContextOptions=array(
+						"ssl"=>array(
+							"verify_peer"=>true,
+							"allow_self_signed"=>true,
```
Comment (DeepDiver1975) [inline lib/Controller/DocumentController.php:170]: just drop all these options please - we shall not allow self signed as wel. THX

File: lib/Controller/DocumentController.php [hunk 3]
```diff
@@ -157,8 +157,21 @@ private function getDiscovery() {
 			$contact_admin = $this->l10n->t('Please contact the "%s" administrator.', [$wopiRemote]);
 
 			try {
-				$wopiClient = \OC::$server->getHTTPClientService()->newClient();
-				$discovery = $wopiClient->get($wopiDiscovery)->getBody();
+				// If we are sending query to built-in CODE server, we avoid using IClint::get() method
```
Comment (phil-davis) [inline lib/Controller/DocumentController.php:160]:
```suggestion
				// If we are sending query to built-in CODE server, we avoid using IClient::get() method
```
Just a small typo in comment

**Comment by mrkara** [comment]
Is there anything I can do to get green marks from codecov? I couldn't find anything in the tests dir, related to the getDiscovery method.

If not, or if that's not a blocker, can we merge this patch? This is prerequitsite for follow-up patches regarding the built-in server.

**Comment by DeepDiver1975** [comment]
> Is there anything I can do to get green marks from codecov? I couldn't find anything in the tests dir, related to the getDiscovery method.

No - this app just lacks any kind of tests :shrug: - not much one can do here

**Comment by mrkara** [comment]
I see. Thank you! :)

**Pull request merged**